Examples/test-suite/same_template_name.i

```diff
@@ -0,0 +1,15 @@
+%module same_template_name;
+
+namespace ns__a {
+
+    template< typename T > class TName {};
+
+}
+
+namespace ns__a {
+    namespace ns__b {
+
+        template< typename T > class TName : public TName<T> {};
+
+    }
+}
```

Comment:
This is not valid C++. I used godbolt.org to check and I get compiler errors with up to date compiler versions: gcc-12.2 and  clang-15.0. Interestingly, icc-2021 and msvc-19.3 did not complain with the code above, but an attempt to instantiate the template with the following then resulted in errors:
```C++
void tester() {
  ns__a::ns__b::TName<int> tn;
}
```
Latest SWIG parses it now without error, but instantiating it with %template does lead to incorrect results. I think that is par for the course.